kotlinx-coroutines-core/common/test/channels/ProduceTest.kt

```diff
@@ -151,7 +151,7 @@ class ProduceTest : TestBase() {
             launch {
                 expect(2)
                 assertFailsWith<IllegalStateException> {
-                    awaitClose { expectUnreached() }
+                    awaitClose { expectUnreached() } // <- access outer ProducerScope from launch scope, won't compile
```

Comment:
Existing code may need to use explicit reference to outer scope in this situation. 
`this@callbackFlow.awaitClose` in this case
Is that expected or is there something I missed?

Reply:
Just to mention that DslMarker will apply to all subclasses including ProducerScope (subclass of CoroutineScope)